cdk/src/open-data-platform/data-plane/schema/schema.sql

```diff
@@ -155,6 +155,120 @@ CREATE TABLE IF NOT EXISTS zipcodes (
+
+-- Precomputed table is required to ensure acceptable latency for the
+-- tileserver.
+CREATE TABLE IF NOT EXISTS demographics_by_county(
+    census_geo_id varchar(255) NOT NULL,
+    name varchar(255) NOT NULL,
+    black_population float,
+    white_population float,
+    total_population real,
+    under_five_population real,
+    poverty_population real,
+    geom geometry(Geometry, 3857),
+    PRIMARY KEY (census_geo_id)
+);
+CREATE INDEX IF NOT EXISTS geom_index ON demographics_by_county USING GIST (geom);
+
+-- Precomputed table is required to ensure acceptable latency for the
+-- tileserver.
+CREATE TABLE IF NOT EXISTS demographics_by_zipcode(
+    census_geo_id varchar(255) NOT NULL,
+    zipcode varchar(255) NOT NULL,
+    black_population float,
+    white_population float,
+    total_population real,
+    under_five_population real,
+    poverty_population real,
+    geom geometry(Geometry, 3857),
+    PRIMARY KEY (census_geo_id)
+);
+CREATE INDEX IF NOT EXISTS geom_index ON demographics_by_zipcode USING GIST (geom);
+
+-- Pre-computed demographic data by state
+-- Only to be used by the function source
+INSERT INTO demographics_by_state(census_geo_id, geom, name, black_population, white_population, total_population, under_five_population, poverty_population)
+SELECT
+    states.census_geo_id as census_geo_id,
+    states.name AS name,
+    ST_Transform(states.geom, 3857) AS geom,
+    SUM(black_population) AS black_population,
+    SUM(white_population) AS white_population,
+    SUM(total_population) AS total_population,
+    SUM(under_five_population) AS under_five_population,
+    SUM(poverty_population) AS poverty_population
+FROM states LEFT JOIN demographics ON demographics.state_census_geo_id = states.census_geo_id
+GROUP BY states.census_geo_id, states.name, states.geom
+ON CONFLICT (census_geo_id) DO NOTHING;
+
+-- Pre-computed demographic data by county
+-- Only to be used by the function source
+INSERT INTO demographics_by_county(census_geo_id, geom, name, black_population, white_population, total_population, under_five_population, poverty_population)
+SELECT
+    counties.census_geo_id as census_geo_id,
+    counties.name AS name,
+    ST_Transform(counties.geom, 3857) AS geom,
+    SUM(black_population) AS black_population,
+    SUM(white_population) AS white_population,
+    SUM(total_population) AS total_population,
+    SUM(under_five_population) AS under_five_population,
+    SUM(poverty_population) AS poverty_population
+FROM counties LEFT JOIN demographics ON demographics.county_census_geo_id = counties.census_geo_id
+GROUP BY counties.census_geo_id, counties.name, counties.geom
+ON CONFLICT (census_geo_id) DO NOTHING;
+
+CREATE OR REPLACE FUNCTION public.demographics_function_source(z integer, x integer, y integer, query_params json) RETURNS bytea AS $$
+DECLARE
+    mvt bytea;
+BEGIN
+    -- If the zoom is low, show state-level demographics. Otherwise, county-level.
```

Comment:
nit, change to: If the zoom is low (i.e. user is zoomed out), show state-level demographics. Otherwise, county-level.

the zoom low thing is confusing to translate mentally I think, so would be nice to clarify here to save that confusion

Reply:
I also found this confusing. Added!